packages/mst-middlewares/test/custom.ts

```diff
@@ -0,0 +1,137 @@
+import { test } from "ava"
+import atomic from "../src/atomic"
+import { types, addMiddleware, getSnapshot } from "mobx-state-tree"
+
+let error: any = null
+
+function omitNextAbort(call, next) {
+    // omit next() / abort()
+}
+function nextAndAbort(call, next, abort) {
+    abort("someValue")
+    next(call)
+}
+
+function abortString(call, next, abort) {
+    abort("someValue")
+}
+function abortNumeric(call, next, abort) {
+    abort(5)
+}
+function nextNoAlter(call, next) {
+    next(call)
+}
+function nextAlter(call, next, abort) {
+    next(call, value => value + 1)
```

Comment:
I don't get this form entirely yet, what is the advantage over 

`return next(call) + 1` ?

Reply:
i guess both would be do able. added this since it separates the actual middleware logic from the value manipulation logic a bit more.

Reply:
I think we should go for "one way of doing things"